client.go

```diff
@@ -29,8 +30,14 @@ func (c noOpCloser) Close() error {
 
 var discard = noOpCloser{w: ioutil.Discard}
 
+type partialJob struct {
+	data     io.WriteCloser
+	warnings io.WriteCloser
```

Comment:
It's a nit, but I'm pretty sure the other one is considered more idiomatic Go. It'd make sense for them to be on separate lines if they were each getting their own comment, but when they're not, it just seems unnecessary to me